server/graphql/programs/src/mutations/rnr_form/update.rs

```diff
@@ -0,0 +1,136 @@
+use async_graphql::*;
+
+use graphql_core::{
+    standard_graphql_error::{validate_auth, StandardGraphqlError},
+    ContextExt,
+};
+use graphql_types::types::rnr_form::RnRFormNode;
+use repository::RnRForm;
+use service::{
+    auth::{Resource, ResourceAccessRequest},
+    rnr_form::update::{UpdateRnRForm, UpdateRnRFormError as ServiceError, UpdateRnRFormLine},
+};
+
+#[derive(InputObject)]
+pub struct UpdateRnRFormInput {
+    pub id: String,
+    pub lines: Vec<UpdateRnRFormLineInput>,
+}
+
+#[derive(InputObject)]
+pub struct UpdateRnRFormLineInput {
+    pub id: String,
+    pub quantity_received: Option<f64>,
+    pub quantity_consumed: Option<f64>,
+    pub adjustments: Option<f64>,
+    pub stock_out_duration: i32,
+    pub adjusted_quantity_consumed: f64,
+    pub final_balance: f64,
+    pub maximum_quantity: f64,
+    pub requested_quantity: f64,
+    pub comment: Option<String>,
+    pub confirmed: bool,
+}
+
+#[derive(Union)]
+pub enum UpdateRnRFormResponse {
+    Response(RnRFormNode),
+}
+
+pub fn update_rnr_form(
+    ctx: &Context<'_>,
+    store_id: String,
+    input: UpdateRnRFormInput,
+) -> Result<UpdateRnRFormResponse> {
+    let user = validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            resource: Resource::MutateRnRForms,
+            store_id: Some(store_id.clone()),
+        },
+    )?;
+    let service_provider = ctx.service_provider();
+    let service_context = service_provider.context(store_id.clone(), user.user_id)?;
+    match service_provider.rnr_form_service.update_rnr_form(
+        &service_context,
+        &store_id,
+        UpdateRnRFormInput::to_domain(input),
+    ) {
+        Ok(RnRForm {
+            rnr_form_row,
+            name_row,
+            store_row: _,
```

Comment:
would be good to pass down and validate that the R&R form belongs to the supplied store